app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,33 @@
+class VotesController < ApplicationController
+  def upvote
+    @review = Review.find(params[:review_id])
+    @user_vote = Vote.find_by(user: current_user)
+    if !@user_vote
+      @review.votes.create(user: current_user, review: @review, upvote: 1, downvote: 0)
+    elsif @user_vote.downvote == 1
+      @user_vote.update_attributes(downvote: 0)
+      @user_vote.update_attributes(upvote: 1)
+    elsif @user_vote.upvote == 1
+      @user_vote.update_attributes(upvote: 0)
+    else
```

Comment:
When will you ever hit this portion of the conditional? I this else scenario is irrelevant and should be taken out (as well as for downvote) unless you have a good reason. This will also help the hound error from above.

Also, this can go either way, especially when taking into account score, but when a value (upvote or downvote) can only be two values (1 or 0), you should consider using a boolean instead. Just my two cents!